File: compiler-rt/lib/msan/lit_tests/select.cc
// RUN: %clangxx_msan -m64 -O0 %s -o %t && not %t >%t.out 2>&1
// RUN: FileCheck %s < %t.out
// RUN: %clangxx_msan -m64 -O1 %s -o %t && not %t >%t.out 2>&1
// RUN: FileCheck %s < %t.out
// RUN: %clangxx_msan -m64 -O2 %s -o %t && not %t >%t.out 2>&1
// RUN: FileCheck %s < %t.out
// RUN: %clangxx_msan -m64 -O3 %s -o %t && not %t >%t.out 2>&1
// RUN: FileCheck %s < %t.out

#include <stdlib.h>
int main(int argc, char **argv) {
  int x;
  int *volatile p = &x;
  int z = *p ? 1 : 0;
  if (z)
    exit(0);
  // CHECK: WARNING: MemorySanitizer: use-of-uninitialized-value
  // CHECK: {{#0 0x.* in main .*select.cc:}}[[@LINE-3]]

  // CHECK: SUMMARY: MemorySanitizer: use-of-uninitialized-value {{.*select.cc:.* main}}
  return 0;
}

File: llgo/test/execution/chan/select.go
// RUN: llgo -o %t %s
// RUN: %t 2>&1 | FileCheck %s

// CHECK: sent a value
// CHECK-NEXT: received 123
// CHECK-NEXT: default

package main

func f1() {
	c := make(chan int, 1)
	for i := 0; i < 3; i++ {
		select {
		case n, _ := <-c:
			println("received", n)
			c = nil
		case c <- 123:
			println("sent a value")
		default:
			println("default")
		}
	}
}

func main() {
	f1()
}
